src/dragonfly_reporter/http_client.py

```diff
@@ -0,0 +1,32 @@
+from collections.abc import Generator
+from functools import cache
+from typing import Annotated
+
+import httpx
+from fastapi import Depends
+
+from dragonfly_reporter.constants import reporter_settings
+
+BASE_URL = "https://pypi.org/danger-api"
+
+
+class BearerAuthentication(httpx.Auth):
+    def __init__(self, *, token: str) -> None:
+        self.token = token
+
+    def auth_flow(
+        self, request: httpx.Request
+    ) -> Generator[httpx.Request, httpx.Response, None]:
+        request.headers["Authorization"] = f"Bearer {self.token}"
+        yield request
+
+
+@cache
+def http_client() -> httpx.AsyncClient:
+    auth = BearerAuthentication(token=reporter_settings.observation_api_token)
+    http_client = httpx.AsyncClient(auth=auth, base_url=BASE_URL)
```

Comment:
This change means we aren't using context managers anymore, is that fine?

Reply:
Yes, though you do bring up a good point. Currently we create the HTTP client but don't close it anywhere. Perhaps this should be done in FastAPI's lifetime.